### DISCUSSION TITLE
16 bit PNG Support

### DISCUSSION THREAD

**Discussion by bukhalmae145** [nothings/stb#1614] [Q&A]
Is 16 bit PNG supported in stb? I'm using stb_write_image.h and stb_image.h with 8 bits. Is there any way to use it in 16 bits with [RealCugan-NCNN-Vulkan](https://github.com/nihui/realcugan-ncnn-vulkan/blob/master/src/main.cpp)?

**Comment by nothings** [comment]
stb_image can read 16-bit png, but stb_image_write is 8-bit only.

**Reply by bukhalmae145** [reply]
Is there any other good encoder/decoder for 16 bit png?

**Reply by nothings** [reply]
`libpng` i would assume

**Reply by p0nce** [reply]
For information I've a modified fork of stb_image_write here that writes 16-bit PNG (with miniz) and a basic heuristic to be a bit more like libpng in size. https://github.com/AuburnSounds/gamut/blob/main/source/gamut/codecs/stb_image_write.d#L268
